src/lib/comms/include/sol-http.h

```diff
@@ -132,7 +135,18 @@ struct sol_http_response {
     int response_code;
 };
 
-#define SOL_HTTP_RESPONSE_CHECK_API(response_, ...) \
+struct sol_http_url {
+    struct sol_str_slice scheme;
+    struct sol_str_slice user;
+    struct sol_str_slice password;
+    struct sol_str_slice host;
+    struct sol_str_slice path;
+    struct sol_str_slice query;
+    struct sol_str_slice fragment;
+    int port; //If < 0 ignore
```

Comment:
On Fri, 06 Nov 2015 09:03:01 -0800, Gustavo Sverzut Barbieri wrote:
> > @@ -132,7 +135,18 @@ struct sol_http_response {
> >      int response_code;
> >  };
> >  
> > -#define SOL_HTTP_RESPONSE_CHECK_API(response_, ...) \
> > +struct sol_http_url {
> > +    struct sol_str_slice scheme;
> > +    struct sol_str_slice user;
> > +    struct sol_str_slice password;
> > +    struct sol_str_slice host;
> > +    struct sol_str_slice path;
> > +    struct sol_str_slice query;
> > +    struct sol_str_slice fragment;
> > +    int port; //If < 0 ignore
> 
> int32_t... and I guess 0 is already not a valid port, so you can make it `uint32_t` and say `if == 0, ignore`

Ports go up to 65535, so uint16_t should be enough, no? But agreed that
int alone is not good, on 16 bit int we lose half the ports.